File: fix.py
import os
import re

files = [
    'ApiKeys.tsx',
    'Auth.tsx',
    'Courses.tsx',
    'IdentityMap.tsx',
    'Settings.tsx',
    'Shortcuts.tsx',
    'Subscriptions.tsx',
    'Vault.tsx'
]

for f in files:
    path = 'D:/PROJEK/saas/vaultpro/frontend/src/pages/' + f
    with open(path, 'r', encoding='utf-8') as file:
        content = file.read()
    
    # Safely re-add backticks
    content = re.sub(r'`?\$\{API_URL\}/api/([a-zA-Z0-9/_-]+)`?', r'`${API_URL}/api/\1`', content)
    
    if f == 'Auth.tsx':
        content = content.replace('`${API_URL}/api/vault`', '`${API_URL}/api/auth/login`')
        # Wait, the register endpoint might also be replaced to auth/login now. I need to be careful.
        # Let's fix Auth.tsx login/register manually in the next step.

    if f == 'IdentityMap.tsx':
        parts = content.split('axios.get(`${API_URL}/api/vault`')
        if len(parts) >= 3:
            content = parts[0] + 'axios.get(`${API_URL}/api/vault`' + parts[1] + 'axios.get(`${API_URL}/api/subscriptions`' + parts[2]
            
        content = content.replace('axios.delete(`${API_URL}/api/vault/${item.id}`', 'axios.delete(`${API_URL}/api/subscriptions/${item.id}`') 
        # Actually I can't guess exactly how it's formatted. I will fix IdentityMap.tsx manually.

    with open(path, 'w', encoding='utf-8') as file:
        file.write(content)

File: fix.js
const fs = require('fs');
const path = require('path');

function replaceInFile(filePath) {
    let content = fs.readFileSync(filePath, 'utf8');
    const original = content;
    
    // Replace broken import.meta stuff
    content = content.replace(/\$\{import\.meta\.env\.VITE_API_URL \|\| 'http:\/\/localhost:5001'\}/g, '${API_URL}/api/vault');
    
    // Now we need to manually fix Auth and Subscriptions
    if (filePath.includes('Auth.tsx')) {
        content = content.replace(/\$\{API_URL\}\/api\/vault/g, '/api/auth/login');
    }
    
    if (filePath.includes('Subscriptions.tsx')) {
        content = content.replace(/\$\{API_URL\}\/api\/vault/g, '/api/subscriptions');
    }
    
    if (filePath.includes('Settings.tsx')) {
        // Settings uses multiple
        // Wait, Settings has GET /api/vault and POST /api/vault/bulk
        // Let's just fix Settings manually
    }
    
    // Replace the remaining template literals http://localhost:5001...
    content = content.replace(/http:\/\/localhost:5001([^]*)/g, '${API_URL}');
    
    // Add import { API_URL } if not exists
    if (content !== original && !content.includes("import { API_URL }")) {
        content = content.replace(/import axios from 'axios';/, "import axios from 'axios';\nimport { API_URL } from '../config';");
    }
    
    fs.writeFileSync(filePath, content, 'utf8');
}

['Auth.tsx', 'Courses.tsx', 'IdentityMap.tsx', 'Settings.tsx', 'Shortcuts.tsx', 'Subscriptions.tsx', 'Vault.tsx'].forEach(f => {
    replaceInFile('D:/PROJEK/saas/vaultpro/frontend/src/pages/' + f);
});

